### PR TITLE
retry when shellnotifyicon fails

### LINKED ISSUE #3693: v3: ShellNotifyIcon fails when shell not loaded yet/properly
### Description

windows system tray calling ShellNotifyIcon with NIM_ADD argument, and panic if syscall fail.
but ShellNotifyIcon with NIM_ADD can be failed when shell(explorer) is not exists yet. especially when app registed as start application via registry

### To Reproduce

1. Having pretty bad computer
2. Register application at start application register key with systray support
3. try reboot until the issue reproduced

### Expected behaviour

retry or let developer handle this type of error

### Screenshots

_No response_

### Attempted Fixes

[I created small patch for this](https://github.com/DeltaLaboratory/wails/commit/51fd0b242348f69cf9b17b2da31254fa0c8e5dff)

### System Details

```shell
^M
```

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by DeltaLaboratory**
# Description
Fixes #3693
I choose 4 for retry and 2 second for delay, no real reason

## Type of change

- [X] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?

- [X] Windows
- [ ] macOS
- [ ] Linux
  
## Test Configuration

Please paste the output of `wails doctor`. If you are unable to run this command, please describe your environment in as much detail as possible.

# Checklist:

- [X] I have updated `website/src/pages/changelog.mdx` with details of this PR
- [ ] My code follows the general coding style of this project
- [X] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [X] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes

## Summary by CodeRabbit

- **New Features**
  - Enhanced the system tray icon setup with a retry mechanism, allowing the application to handle temporary failures more gracefully.

- **Bug Fixes**
  - Added a changelog entry addressing a critical system tray startup issue on Windows, preventing unexpected application crashes.

### CODE REVIEW

File: v3/pkg/application/systemtray_windows.go [hunk 1]
```diff
@@ -174,8 +176,16 @@ func (s *windowsSystemTray) run() {
 	}
 	nid.CbSize = uint32(unsafe.Sizeof(nid))
 
-	if !w32.ShellNotifyIcon(w32.NIM_ADD, &nid) {
-		panic(syscall.GetLastError())
+	for retries := range 4 {
+		if !w32.ShellNotifyIcon(w32.NIM_ADD, &nid) {
+			if retries == 3 {
+				panic(syscall.GetLastError())
```
[the remark this answers is not in the document]
Comment (leaanthony) [reply v3/pkg/application/systemtray_windows.go:182]: @DeltaLaboratory - I think this is a valid comment from the rabbit.

File: v3/pkg/application/systemtray_windows.go [hunk 2]
```diff
@@ -174,8 +176,16 @@
...
+			}
+
+			time.Sleep(2 * time.Second)
+			continue
+		}
+		break
```
[the remark this answers is not in the document]
Comment (leaanthony) [reply v3/pkg/application/systemtray_windows.go:188]: We aren't doing this

**Comment by leaanthony** [comment]
Thanks for taking the time to open this. It would be better to use smaller sleeps and a higher number of attempts. I'm concerned this is a blocking operation so it would be better to minimise the delay here.

File: v3/pkg/application/systemtray_windows.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (DeltaLaboratory) [reply v3/pkg/application/systemtray_windows.go:182]: @leaanthony I think this is a good idea, but in the current codebase how can we modify this to allow user to handle error?
```go
func (s *SystemTray) Run() {
	s.impl = newSystemTrayImpl(s)

	if s.attachedWindow.Window != nil {
		// Setup listener
		s.attachedWindow.Window.On(events.Common.WindowLostFocus, func(event *WindowEvent) {
			s.attachedWindow.Window.Hide()
			// Special handler for Windows
			if runtime.GOOS == "windows" {
				// We don't do this unless the window has already been shown
				if s.attachedWindow.hasBeenShown == false {
					return
				}
				s.attachedWindow.justClosed = true
				go func() {
					time.Sleep(s.attachedWindow.Debounce)
					s.attachedWindow.justClosed = false
				}()
			}
		})
	}

	InvokeSync(s.impl.run)
}```

**Comment by DeltaLaboratory** [comment]
I checked out some Windows startup times and noticed there's usually about a 3-second gap between when programs start loading and when the desktop is ready to go. So, I adjusted the settings to wait 500 milliseconds and try 6 times.

**Comment by leaanthony** [comment]
Thanks 🙏 Please could you add an entry to the changelog located at `mkdocs-website/docs/changelog.md`? Thanks!

**Comment by DeltaLaboratory** [comment]
added changelog

**Comment by DeltaLaboratory** [comment]
hello, is there anything blocking this merge?

**Comment by leaanthony** [comment]
Just time :-)

File: v3/pkg/application/systemtray_windows.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (leaanthony) [reply v3/pkg/application/systemtray_windows.go]: Pushed an update.

**Review by leaanthony — APPROVED** [review]

**Pull request merged**